companion/lib/Service/HttpApi.ts

```diff
@@ -641,4 +649,164 @@ export class ServiceHttpApi {
 			}
 		}
 	}
+
+	/**
+	 * Provides JSON appinfo
+	 *
+	 */
+	#appInfoGetJson = (_req: Express.Request, res: Express.Response): void => {
+		this.logger.debug(`Got HTTP /api/appinfo/json`)
+		res.json({
+			success: true,
+			message: 'App information retrieved successfully',
+			timestamp: getTimestampUnix(),
+			data: {
+				appinfo: this.#serviceApi.appInfo,
+			},
+		})
+		return
+	}
+
+	/**
+	 * Provides JSON for connection status
+	 *
+	 */
+	#connectionGetJson = (_req: Express.Request, res: Express.Response): void => {
+		this.logger.debug(`Got HTTP /api/connections/json`)
+
+		const internalVariables = this.#serviceApi.getConnectionVariableDefinitions('internal')
+
+		const connections = Object.entries(internalVariables)
+			.filter(([key]) => key.startsWith('connection_') && key.endsWith('_status'))
+			.map(([key]) => ({
+				name: key.replace('connection_', '').replace('_status', ''),
+				status: this.#serviceApi.getConnectionVariableValue('internal', key),
+			}))
+
+		const instanceStatus = Object.entries(internalVariables)
+			.filter(([key]) => key.startsWith('instance_'))
+			.reduce(
+				(acc, [key]) => ({
+					...acc,
+					[key.replace('instance_', '')]: {
+						count: this.#serviceApi.getConnectionVariableValue('internal', key),
+					},
+				}),
+				{}
+			)
+
+		res.json({
+			success: true,
+			message: 'Connection information retrieved successfully',
+			timestamp: getTimestampUnix(),
+			data: {
+				connections,
+				instanceStatus,
+			},
+		})
+		return
+	}
+
+	/**
+	 * Provides JSON for module or custom variables
+	 *
+	 */
+	#variablesGetJson = (req: Express.Request, res: Express.Response): void => {
+		const connectionLabel = req.params.label
+		this.logger.debug(`Got HTTP /api/variables/${connectionLabel}/json`)
+
+		// Determine variable type and fetch definitions
+		const isCustomVariable = connectionLabel === 'custom'
+		const variableDefinitions = isCustomVariable
+			? this.#serviceApi.getCustomVariableDefinitions()
+			: this.#serviceApi.getConnectionVariableDefinitions(connectionLabel)
+
+		const result: Record<string, any> = {}
+
+		for (const [variableName, obj] of Object.entries(variableDefinitions)) {
+			const value = isCustomVariable
+				? this.#serviceApi.getCustomVariableValue(variableName)
+				: this.#serviceApi.getConnectionVariableValue(connectionLabel, variableName)
+
+			if (value !== undefined) {
+				result[variableName] = {
```

Comment:
I don't know the right answer here, but in the Prometheus endpoint, you've included the connection name too.

If I've got multiple connections, talking to the same type of device (e.g. 3 Hyperdecks from the original issue), without the "namespace" being embedded in the returned JSON, I've got to do some processing before I can store them (or store by "namespace"), whereas I could just chuck all the Prometheus datapoints in a file and they're still unique. Although I appreciate it's more data being transmitted that's duplicated if I'm using them separately.